---
title: "SEESAW: Statistical Estimation Of Allelic Expression Using Salmon And Swish"
author: Euphy Wu, Michael Love
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{SEESAW: Statistical Estimation Of Allelic Expression Using Salmon And Swish}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Allelic Expression Analysis

Authors:
    Euphy Wu^[UNC-Chapel Hill],
    Michael Love^[UNC-Chapel Hill].
    <br/>
Last modified: July, 2022.

## Overview

### Description

In this workshop, we will cover a newly developed pipeline for
performing analysis of allelic expression in Bioconductor. Students
will learn about how we use upstream tools like `g2gtools` and
`Salmon` to quantify allelic expression, and then they will be able to
explore the data in R using the `fishpond` package. 

### Pre-requisites

* Some familiarity with RNA-seq data will help
* Some familiarity with working with *SummarizedExperiment* objects will help

Theoretical background on allelic expression:

* [Tools and best practices for data processing in allelic expression analysis](https://doi.org/10.1186/s13059-015-0762-6)
* [A vast resource of allelic expression data spanning human tissues](https://doi.org/10.1186/s13059-020-02122-z)
* [Hierarchical analysis of RNA-seq reads improves the accuracy of allele-specific expression](https://doi.org/10.1093/bioinformatics/bty078)
* [Haplotype and isoform specific expression estimation using multi-mapping RNA-seq reads](https://doi.org/10.1186/gb-2011-12-2-r13)

### Participation

The format is lecture + lab, with plenty of time for data exploration.

### _R_ / _Bioconductor_ packages used

* SummarizedExperiment
* fishpond
* plyranges
* pheatmap
* Gviz

### Time outline

An example for a 45-minute workshop:

| Activity                     | Time |
|------------------------------|------|
| Intro to allelic quant       | 15m  |
| Exploration of data in R     | 15m  |
| Testing and visualization    | 15m  |

### Workshop goals and objectives

### Learning goals

* describe how to...
* identify methods for...
* understand the difference between...

### Learning objectives

* analyze xyz data to produce...
* create xyz plots
* evaluate xyz data for artifacts

## SEESAW pipeline

At the beginning of the workshop, we will give an overview of the
steps that need to take place outside of R, before we import the
allelic data into Bioconductor. These are outlined as the first parts
of the pipeline in the diagram below. These are also described in
detail in the 
[allelic vignette](https://mikelove.github.io/fishpond/articles/allelic.html) 
for the fishpond package.

```{r, echo=FALSE}
knitr::include_graphics("images/SEESAW.png")
```

Links to relevant software: 

* [g2gtools](http://churchill-lab.github.io/g2gtools/) for
  constructing the diploid transcriptome
* Salmon, run with [bootstrap inferential replicates](https://salmon.readthedocs.io/en/latest/salmon.html#numbootstraps)

## Importing allelic counts into R

The following three code chunks were used to import the allelic
quantification from Salmon into R. These are un-evaluated, as the
original data is too large to include in this package.

First, we define `coldata` and `txps`, which describe the sample
metadata, and the genomic ranges of the transcripts that were
quantified (the haploid reference transcripts). We do not yet have a
convenient way to describe the location of the transcripts in the
diploid genome, so we still rely on a single reference genome here,
although a diploid genome was used for quantifying allelic expression.

```{r eval=FALSE}
# `files` points to `quant.sf` files in original Salmon directories
# other vectors here describe the samples
coldata <- data.frame(cross, day, files, names)
ah <- AnnotationHub()
#query(ah, c("EnsDb","102","Mus musculus"))
edb <- ah[["AH89211"]]
txps <- transcripts(edb)
```

The following code chunk imports the allelic data, creating a "wide"
*SummarizedExperiment* object which will have 2x the number of columns
as the samples listed in `coldata`. We specify the `a1` and `a2`
allele. By convention, the `a2` allele describes the reference allele.

```{r eval=FALSE}
library(fishpond)
gse <- importAllelicCounts(
  coldata, a1="alt", a2="ref",
  format="wide", tx2gene=txps,
)
```

Finally, we performed minimal filtering on the features:

```{r eval=FALSE}
# filtering out lowly expressed features:
keep <- rowSums(assay(gse) >= 10) >= 6
table(keep)
gse <- gse[keep,]
```

```{r message=FALSE}
library(Bioc2022AllelicExpression)
library(SummarizedExperiment)
data("osteoblast_gene_chr1")
gene
```
