---
title: "SEESAW: Statistical Estimation Of Allelic Expression Using Salmon And Swish"
author: Euphy Wu, Michael Love
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{SEESAW: Statistical Estimation Of Allelic Expression Using Salmon And Swish}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Allelic Expression Analysis

Authors:
    Euphy Wu^[UNC-Chapel Hill],
    Michael Love^[UNC-Chapel Hill].
    <br/>
Last modified: July, 2022.

## Overview

### Description

Along with the topic of your workshop, include how students can expect
to spend their time. For the description may also include information
about what type of workshop it is (e.g. instructor-led live demo, lab,
lecture + lab, etc.). Instructors are strongly recommended to provide
completely worked examples for lab sessions, and a set of stand-alone
notes that can be read and understood outside of the workshop.

### Pre-requisites

List any workshop prerequisites, for example:

* Basic knowledge of R syntax
* Familiarity with the GenomicRanges class
* Familiarity with xyz vignette (provide link)

List relevant background reading for the workshop, including any
theoretical background you expect students to have.

* List any textbooks, papers, or other reading that students should be
  familiar with. Include direct links where possible.

### Participation

Describe how students will be expected to participate in the workshop.

### _R_ / _Bioconductor_ packages used

* SummarizedExperiment
* fishpond
* plyranges
* pheatmap
* Gviz

### Time outline

An example for a 45-minute workshop:

| Activity                     | Time |
|------------------------------|------|
|                              | 15m  |
|                              | 15m  |
|                              | 15m  |

### Workshop goals and objectives

List "big picture" student-centered workshop goals and learning
objectives. Learning goals and objectives are related, but not the
same thing. These goals and objectives will help some people to decide
whether to attend the conference for training purposes, so please make
these as precise and accurate as possible.

*Learning goals* are high-level descriptions of what
participants will learn and be able to do after the workshop is
over. *Learning objectives*, on the other hand, describe in very
specific and measurable terms specific skills or knowledge
attained. The [Bloom's Taxonomy](#bloom) may be a useful framework
for defining and describing your goals and objectives, although there
are others.

### Learning goals

Some examples:

* describe how to...
* identify methods for...
* understand the difference between...

### Learning objectives

* analyze xyz data to produce...
* create xyz plots
* evaluate xyz data for artifacts

## SEESAW pipeline

At the beginning of the workshop, we will give an overview of the
steps that need to take place outside of R, before we import the
allelic data into Bioconductor. These are outlined as the first parts
of the pipeline in the diagram below. These are also described in
detail in the 
[allelic vignette](https://mikelove.github.io/fishpond/articles/allelic.html) 
for the fishpond package.

```{r, echo=FALSE}
knitr::include_graphics("images/seesaw.png")
```

Links to relevant software: 

* [g2gtools](http://churchill-lab.github.io/g2gtools/) for
  constructing the diploid transcriptome
* Salmon, run with [bootstrap inferential replicates](https://salmon.readthedocs.io/en/latest/salmon.html#numbootstraps)

## Importing allelic counts into R

The first goes here...
