---
title: "SEESAW: Statistical Estimation Of Allelic Expression Using Salmon And Swish"
author: Euphy Wu, Michael Love
date: "July 2022"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{SEESAW: Statistical Estimation Of Allelic Expression Using Salmon And Swish}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Overview

### Description

In this workshop, we will cover a newly developed suite of tools for
performing analysis of allelic expression in Bioconductor. You
will learn about how we use upstream tools like `g2gtools` and
`Salmon` to quantify allelic expression, and then how to explore the
data in R using the `fishpond` package. Altogether we call this suite
of tools *SEESAW* for
"Statistical Estimation Of allelic Expression Using Salmon And sWish".

This set of methods for quantifying and analyzing allelic data is
similar to 
[mmseq](https://github.com/eturro/mmseq), 
in which uncertainty of assignment of reads to isoforms and alleles is
estimated, stored, and using in statistical testing.
It is also similar to 
[EMASE](https://emase.readthedocs.io/en/latest/index.html), which is a
quantification engine for isoform- and allele-level quantification
(we also use the same upstream pipeline for construction of diploid
reference transcripts).
Our approach differs from other methods such as 
[WASP](https://github.com/bmvdgeijn/WASP)
or the Bioconductor package
[AllelicImbalance](https://www.bioconductor.org/packages/AllelicImbalance),
in that we do not focus on pileup of reads on particular SNPs, but
instead integrate the isoform- and allele-level information contained
in reads aligning to entire transcripts (including both SNP and indel
variation). 

### Pre-requisites

* Some basic background of RNA biology, e.g. what are genes, what are isoforms/transcripts?
* Some familiarity with RNA-seq data, e.g. what are reads?
* Some familiarity with working with *SummarizedExperiment* objects will help.

Theoretical background on allelic expression:

* [Tools and best practices for data processing in allelic expression analysis](https://doi.org/10.1186/s13059-015-0762-6)
* [A vast resource of allelic expression data spanning human tissues](https://doi.org/10.1186/s13059-020-02122-z)
* [Hierarchical analysis of RNA-seq reads improves the accuracy of allele-specific expression](https://doi.org/10.1093/bioinformatics/bty078)
* [Haplotype and isoform specific expression estimation using multi-mapping RNA-seq reads](https://doi.org/10.1186/gb-2011-12-2-r13)

### Participation

The format is lecture + lab, with plenty of time for data exploration.

### _R_ / _Bioconductor_ packages used

* SummarizedExperiment
* fishpond
* plyranges
* pheatmap
* Gviz

### Time outline

An example for a 45-minute workshop:

| Activity                     | Time |
|------------------------------|------|
| Intro to allelic quant       | 15m  |
| Exploration of data in R     | 15m  |
| Testing and visualization    | 15m  |

### Workshop goals and objectives

### Learning goals

* describe how to...
* identify methods for...
* understand the difference between...

### Learning objectives

* analyze xyz data to produce...
* create xyz plots
* evaluate xyz data for artifacts

## What is allelic expression and why do we care

...

## SEESAW pipeline

At the beginning of the workshop, we will give an overview of the
steps that need to take place outside of R, before we import the
allelic data into Bioconductor. These are outlined as the first parts
of the pipeline in the diagram below. These are also described in
detail in the 
[allelic vignette](https://mikelove.github.io/fishpond/articles/allelic.html) 
for the fishpond package.

```{r, echo=FALSE}
knitr::include_graphics("images/SEESAW.png")
```

Links to relevant software: 

* [g2gtools](http://churchill-lab.github.io/g2gtools/) for
  constructing the diploid transcriptome
* Salmon, run with [bootstrap inferential replicates](https://salmon.readthedocs.io/en/latest/salmon.html#numbootstraps)

## Importing allelic counts

The following five un-evaluated code chunks were used to import the
allelic quantification from Salmon into R. These are un-evaluated, as
the original data is too large to include in this package.

First, we define `coldata` and `txps`, which describe the sample
metadata, and the genomic ranges of the transcripts that were
quantified (the haploid reference transcripts). We do not yet have a
convenient way to describe the location of the transcripts in the
diploid genome, so we still rely on a single reference genome here,
although a diploid genome was used for quantifying allelic expression.

```{r eval=FALSE}
# `files` points to `quant.sf` files in original Salmon directories
# other vectors here describe the samples
coldata <- data.frame(cross, day, files, names)
ah <- AnnotationHub()
#query(ah, c("EnsDb","102","Mus musculus"))
edb <- ah[["AH89211"]]
txps <- transcripts(edb)
```

We can group to the gene level with the following `group_id`:

```{r eval=FALSE}
library(plyranges)
txps <- txps %>%
  select(tx_id, group_id=gene_id)
```

The following code chunk imports the allelic data, creating a "wide"
*SummarizedExperiment* object which will have 2x the number of columns
as the samples listed in `coldata`. We specify the strings that
distinguish the `a1` and `a2` allele in the transcript FASTA file 
(these are suffices that follow the transcript name, separated with an
underscore, e.g. `ENST123_ref` and `ENST123_alt`.
By convention, the `a2` allele describes the reference allele.
In the `gse` object (gene-level SummarizedExperiment), the `a2`
allelic counts will appear as the first columns followed by the `a1`
allelic counts. Here we supply `txps` as a *GRanges* object with a
metadata column `group_id` that describes how we collapse
transcript-level counts.


```{r eval=FALSE}
library(fishpond)
gse <- importAllelicCounts(
  coldata, a1="alt", a2="ref",
  format="wide", tx2gene=txps,
)
```

Finally, we performed minimal filtering on the features:

```{r eval=FALSE}
# filtering out lowly expressed features:
keep <- rowSums(assay(gse) >= 10) >= 6
table(keep)
gse <- gse[keep,]
```

Alternatively, we can group transcript-level counts to the TSS level
using the following code chunk to define `txps`. `makeTx2Tss()` is a
convenience function in *fishpond*. The `maxgap=50` argument means we
will group together transcripts that have TSS that fall within 50bp of
each other.

```{r eval=FALSE}
txps <- makeTx2Tss(edb, maxgap=50) %>%
  select(tx_id, gene_id, group_id, tss)
```

## Explore allelic counts

We have two pre-packaged datasets as part of this workshop
package, one where allelic counts from an RNA-seq experiment are
summarized to the gene level, and another where they are summarized to
the TSS level.

The datasets are described in the man pages (see References), but
briefly, mouse osteoblast cells were differentiated over a time course
from day 2 to day 18. The data has been subset to just the genes on
chr1. A reference for the experiment is:

> Kemp JP, Medina-Gomez C, Estrada K, St Pourcain B et al.
> Phenotypic dissection of bone mineral density reveals skeletal site specificity
> and facilitates the identification of novel loci in the genetic regulation of
> bone mass attainment. PLoS Genet 2014 Jun;10(6):e1004423. PMID: 24945404
> <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4063697/>

We load the gene-level dataset:

```{r message=FALSE}
library(SummarizedExperiment)
library(Bioc2022AllelicExpression)
data(osteoblast_gene_chr1)
osteoblast_gene_chr1
g <- osteoblast_gene_chr1
```

The sample metadata:

```{r}
colData(g)
```

For each sample and each allele, we have counts and also bootstrap
replicates. The bootstrap replicates are from Salmon, by specifying
`--numBootstraps 30` during quantification.

We can use *fishpond* functions to do some basic exploration of the
allelic counts. 

```{r histBoot}
library(fishpond)
assayNames(g)
dat <- getTrace(g, idx=1, samp_idx="129xB6-d02-a2")
hist(dat$count, border="white", col="grey50", xlab="estimated count, a2")
```

This plot shows the distribution of estimated counts for the first
feature, and for the day 2 sample of the 129xB6 cross, reference allele.
We can also visualize the distributions for all samples using
`plotInfReps()`:

```{r plotInfReps}
plotInfReps(g, idx=1, x="allele", legend=TRUE)
```

We can also group the allelic counts by sample, e.g.:

```{r plotInfReps2}
plotInfReps(g[,g$cross == "CASTxB6"],
            idx=1, x="allele", cov="day",
            legend=TRUE)
```
