<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SEESAW: Statistical Estimation Of Allelic Expression Using Salmon And Swish • Bioc2022AllelicExpression</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="SEESAW: Statistical Estimation Of Allelic Expression Using Salmon And Swish">
<meta property="og:description" content="Bioc2022AllelicExpression">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Bioc2022AllelicExpression</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/allelicExpression.html">SEESAW: Statistical Estimation Of Allelic Expression Using Salmon And Swish</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mikelove/Bioc2022AllelicExpression" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>SEESAW: Statistical Estimation Of Allelic
Expression Using Salmon And Swish</h1>
                        <h4 data-toc-skip class="author">Euphy Wu,
Michael Love</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mikelove/Bioc2022AllelicExpression/blob/HEAD/vignettes/allelicExpression.Rmd" class="external-link"><code>vignettes/allelicExpression.Rmd</code></a></small>
      <div class="hidden name"><code>allelicExpression.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="allelic-expression-analysis">Allelic Expression Analysis<a class="anchor" aria-label="anchor" href="#allelic-expression-analysis"></a>
</h2>
<p>Authors: Euphy Wu<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>, Michael Love<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>. <br> Last modified:
July, 2022.</p>
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<div class="section level4">
<h4 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h4>
<p>In this workshop, we will cover a newly developed suite of tools for
performing analysis of allelic expression in Bioconductor. Students will
learn about how we use upstream tools like <code>g2gtools</code> and
<code>Salmon</code> to quantify allelic expression, and then they will
be able to explore the data in R using the <code>fishpond</code>
package. Altogether we call this suite of tools <em>SEESAW</em> for
“Statistical Estimation Of allelic Expression Using Salmon And
sWish”.</p>
<p>This set of methods for quantifying and analyzing allelic data is
similar to <a href="https://github.com/eturro/mmseq" class="external-link">mmseq</a>, in which
uncertainty of assignment of reads to isoforms and alleles is estimated,
stored, and using in statistical testing. It is also similar to <a href="https://emase.readthedocs.io/en/latest/index.html" class="external-link">EMASE</a>,
which is a quantification engine for isoform- and allele-level
quantification (we also use the same upstream pipeline for construction
of diploid reference transcripts). Our approach differs from other
methods such as <a href="https://github.com/bmvdgeijn/WASP" class="external-link">WASP</a> or
the Bioconductor package <a href="https://www.bioconductor.org/packages/AllelicImbalance" class="external-link">AllelicImbalance</a>,
in that we do not focus on pileup of reads on particular SNPs, but
instead integrate the isoform- and allele-level information contained in
reads aligning to entire transcripts (including both SNP and indel
variation).</p>
</div>
<div class="section level4">
<h4 id="pre-requisites">Pre-requisites<a class="anchor" aria-label="anchor" href="#pre-requisites"></a>
</h4>
<ul>
<li>Some familiarity with RNA-seq data will help</li>
<li>Some familiarity with working with <em>SummarizedExperiment</em>
objects will help</li>
</ul>
<p>Theoretical background on allelic expression:</p>
<ul>
<li><a href="https://doi.org/10.1186/s13059-015-0762-6" class="external-link">Tools and best
practices for data processing in allelic expression analysis</a></li>
<li><a href="https://doi.org/10.1186/s13059-020-02122-z" class="external-link">A vast resource
of allelic expression data spanning human tissues</a></li>
<li><a href="https://doi.org/10.1093/bioinformatics/bty078" class="external-link">Hierarchical
analysis of RNA-seq reads improves the accuracy of allele-specific
expression</a></li>
<li><a href="https://doi.org/10.1186/gb-2011-12-2-r13" class="external-link">Haplotype and
isoform specific expression estimation using multi-mapping RNA-seq
reads</a></li>
</ul>
</div>
<div class="section level4">
<h4 id="participation">Participation<a class="anchor" aria-label="anchor" href="#participation"></a>
</h4>
<p>The format is lecture + lab, with plenty of time for data
exploration.</p>
</div>
<div class="section level4">
<h4 id="r-bioconductor-packages-used">
<em>R</em> / <em>Bioconductor</em> packages used<a class="anchor" aria-label="anchor" href="#r-bioconductor-packages-used"></a>
</h4>
<ul>
<li>SummarizedExperiment</li>
<li>fishpond</li>
<li>plyranges</li>
<li>pheatmap</li>
<li>Gviz</li>
</ul>
</div>
<div class="section level4">
<h4 id="time-outline">Time outline<a class="anchor" aria-label="anchor" href="#time-outline"></a>
</h4>
<p>An example for a 45-minute workshop:</p>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Intro to allelic quant</td>
<td>15m</td>
</tr>
<tr class="even">
<td>Exploration of data in R</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>Testing and visualization</td>
<td>15m</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="workshop-goals-and-objectives">Workshop goals and objectives<a class="anchor" aria-label="anchor" href="#workshop-goals-and-objectives"></a>
</h4>
</div>
<div class="section level4">
<h4 id="learning-goals">Learning goals<a class="anchor" aria-label="anchor" href="#learning-goals"></a>
</h4>
<ul>
<li>describe how to…</li>
<li>identify methods for…</li>
<li>understand the difference between…</li>
</ul>
</div>
<div class="section level4">
<h4 id="learning-objectives">Learning objectives<a class="anchor" aria-label="anchor" href="#learning-objectives"></a>
</h4>
<ul>
<li>analyze xyz data to produce…</li>
<li>create xyz plots</li>
<li>evaluate xyz data for artifacts</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="seesaw-pipeline">SEESAW pipeline<a class="anchor" aria-label="anchor" href="#seesaw-pipeline"></a>
</h3>
<p>At the beginning of the workshop, we will give an overview of the
steps that need to take place outside of R, before we import the allelic
data into Bioconductor. These are outlined as the first parts of the
pipeline in the diagram below. These are also described in detail in the
<a href="https://mikelove.github.io/fishpond/articles/allelic.html" class="external-link">allelic
vignette</a> for the fishpond package.</p>
<p><img src="images/SEESAW.png"><!-- --></p>
<p>Links to relevant software:</p>
<ul>
<li>
<a href="http://churchill-lab.github.io/g2gtools/" class="external-link">g2gtools</a> for
constructing the diploid transcriptome</li>
<li>Salmon, run with <a href="https://salmon.readthedocs.io/en/latest/salmon.html#numbootstraps" class="external-link">bootstrap
inferential replicates</a>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="importing-allelic-counts-into-r">Importing allelic counts into R<a class="anchor" aria-label="anchor" href="#importing-allelic-counts-into-r"></a>
</h3>
<p>The following three code chunks were used to import the allelic
quantification from Salmon into R. These are un-evaluated, as the
original data is too large to include in this package.</p>
<p>First, we define <code>coldata</code> and <code>txps</code>, which
describe the sample metadata, and the genomic ranges of the transcripts
that were quantified (the haploid reference transcripts). We do not yet
have a convenient way to describe the location of the transcripts in the
diploid genome, so we still rely on a single reference genome here,
although a diploid genome was used for quantifying allelic
expression.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># `files` points to `quant.sf` files in original Salmon directories</span></span>
<span><span class="co"># other vectors here describe the samples</span></span>
<span><span class="va">coldata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">cross</span>, <span class="va">day</span>, <span class="va">files</span>, <span class="va">names</span><span class="op">)</span></span>
<span><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#query(ah, c("EnsDb","102","Mus musculus"))</span></span>
<span><span class="va">edb</span> <span class="op">&lt;-</span> <span class="va">ah</span><span class="op">[[</span><span class="st">"AH89211"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">txps</span> <span class="op">&lt;-</span> <span class="fu">transcripts</span><span class="op">(</span><span class="va">edb</span><span class="op">)</span></span></code></pre></div>
<p>The following code chunk imports the allelic data, creating a “wide”
<em>SummarizedExperiment</em> object which will have 2x the number of
columns as the samples listed in <code>coldata</code>. We specify the
<code>a1</code> and <code>a2</code> allele. By convention, the
<code>a2</code> allele describes the reference allele.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/fishpond" class="external-link">fishpond</a></span><span class="op">)</span></span>
<span><span class="va">gse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/fishpond/man/importAllelicCounts.html" class="external-link">importAllelicCounts</a></span><span class="op">(</span></span>
<span>  <span class="va">coldata</span>, a1<span class="op">=</span><span class="st">"alt"</span>, a2<span class="op">=</span><span class="st">"ref"</span>,</span>
<span>  format<span class="op">=</span><span class="st">"wide"</span>, tx2gene<span class="op">=</span><span class="va">txps</span>,</span>
<span><span class="op">)</span></span></code></pre></div>
<p>Finally, we performed minimal filtering on the features:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filtering out lowly expressed features:</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">gse</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">6</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">keep</span><span class="op">)</span></span>
<span><span class="va">gse</span> <span class="op">&lt;-</span> <span class="va">gse</span><span class="op">[</span><span class="va">keep</span>,<span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mikelove.github.io/Bioc2022AllelicExpression">Bioc2022AllelicExpression</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">osteoblast_gene_chr1</span><span class="op">)</span></span>
<span><span class="va">osteoblast_gene_chr1</span></span>
<span><span class="co">#&gt; class: RangedSummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 1211 36 </span></span>
<span><span class="co">#&gt; metadata(3): tximetaInfo countsFromAbundance alleles</span></span>
<span><span class="co">#&gt; assays(33): counts abundance ... infRep29 infRep30</span></span>
<span><span class="co">#&gt; rownames(1211): ENSMUSG00000001138 ENSMUSG00000001143 ...</span></span>
<span><span class="co">#&gt;   ENSMUSG00000118219 ENSMUSG00000118607</span></span>
<span><span class="co">#&gt; rowData names(9): gene_id gene_name ... symbol entrezid</span></span>
<span><span class="co">#&gt; colnames(36): 129xB6-d02-a2 129xB6-d04-a2 ... CASTxB6-d16-a1</span></span>
<span><span class="co">#&gt;   CASTxB6-d18-a1</span></span>
<span><span class="co">#&gt; colData names(3): allele cross day</span></span></code></pre></div>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p>UNC-Chapel Hill<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>UNC-Chapel Hill<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Euphy Wu, Michael Love.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
